src/Mocking.jl

```diff
@@ -40,17 +40,17 @@ struct Patch{T}
 end
 
 macro patch(expr::Expr)
+    def = splitdef(expr, throw=false)
+
     # Expect a named function in long-form or short-form
-    if expr.head === :function || (expr.head === :(=) && expr.args[1].head == :call)
-        target_name = expr.args[1].args[1]
-        params = expr.args[1].args[2:end]
-        body = expr.args[2]
-    elseif expr.head == :(->)
-        throw(ArgumentError("expression needs to be a named function"))
-    else
+    if def === nothing
         throw(ArgumentError("expression is not a function definition"))
```

Comment:
Why not make `splitdef` throw the error?

Reply:
I think originally I didn't have the `splitdef` exception throwing working very well. At this point I think I can just have `splitdef` throw the exception.

Reply:
Oh, I may have been playing with the idea of returning the exception and letting it fail at runtime instead of during macro expansion